src/System.Numerics.Matrices/src/System/Numerics/Matrix4x4.cs

```diff
@@ -0,0 +1,435 @@
+
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 1
+        /// </summary>
+        public Matrix1x4 Column1 { get { return new Matrix1x4(M11, M12, M13, M14); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 2
+        /// </summary>
+        public Matrix1x4 Column2 { get { return new Matrix1x4(M21, M22, M23, M24); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 3
+        /// </summary>
+        public Matrix1x4 Column3 { get { return new Matrix1x4(M31, M32, M33, M34); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 4
+        /// </summary>
+        public Matrix1x4 Column4 { get { return new Matrix1x4(M41, M42, M43, M44); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 1
+        /// </summary>
+        public Matrix4x1 Row1 { get { return new Matrix4x1(M11, M21, M31, M41); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 2
+        /// </summary>
+        public Matrix4x1 Row2 { get { return new Matrix4x1(M12, M22, M32, M42); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 3
+        /// </summary>
+        public Matrix4x1 Row3 { get { return new Matrix4x1(M13, M23, M33, M43); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 4
+        /// </summary>
+        public Matrix4x1 Row4 { get { return new Matrix4x1(M14, M24, M34, M44); } }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Matrix4x4)
+                return this == (Matrix4x4)obj;
+
+            return false;
+        }
+
+        public bool Equals(Matrix4x4 other)
+        {
+            return this == other;
+        }
+
+        public unsafe override int GetHashCode()
+        {
+            fixed (Matrix4x4* p = &this)
+            {
+                int* x = (int*)p;
+                unchecked
+                {
+                    return 0xFFE1
+                         + 7 * ((((x[00] ^ x[01]) << 0) + ((x[02] ^ x[03]) << 1) + ((x[04] ^ x[05]) << 2) + ((x[06] ^ x[07]) << 3)) << 0)
+                         + 7 * ((((x[04] ^ x[05]) << 0) + ((x[06] ^ x[07]) << 1) + ((x[08] ^ x[09]) << 2) + ((x[10] ^ x[11]) << 3)) << 1)
+                         + 7 * ((((x[08] ^ x[09]) << 0) + ((x[10] ^ x[11]) << 1) + ((x[12] ^ x[13]) << 2) + ((x[14] ^ x[15]) << 3)) << 2)
+                         + 7 * ((((x[12] ^ x[13]) << 0) + ((x[14] ^ x[15]) << 1) + ((x[16] ^ x[17]) << 2) + ((x[18] ^ x[19]) << 3)) << 3);
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            return "Matrix4x4: "
+                 + $"{{|{M11:00}|{M21:00}|{M31:00}|{M41:00}|}}"
+                 + $"{{|{M12:00}|{M22:00}|{M32:00}|{M42:00}|}}"
```

Comment:
I originally did this with String.Format(...) and having the concat strings was "free" as they should have been concatenated at compile time. With interpolated strings, does it mean a bunch of strings will get alloc'd then concatenated on each ToString invocation?